schema/crdb/dbinit.sql

```diff
@@ -202,6 +201,7 @@ CREATE TYPE IF NOT EXISTS omicron.public.service_kind AS ENUM (
   'ntp',
   'oximeter',
   'tfport',
+  'clickhouse_keeper',
```

Comment:
All this is to say -- we **can** make this change, and if you want, but I think it's still unsafe to rely on order of our enums with possible deployments unless we add a "schema change" that validates order.

Reply:
(also, to be explicit, the changes that you've added to the integration tests are extremely useful still, in either situation)

Reply:
It's a pain-in-the-ass, but we could:

- Add a new enum variant to service_kind -- at the end -- called clickhouse_keeper2
- Update omicron.public.service_kind usage, anywhere that clickhouse_keeper is used, update the index to clickhouse_keeper2
- Drop the clickhouse_keeper variant (see: https://www.cockroachlabs.com/docs/v23.1/alter-type#drop-a-value-in-a-user-defined-type)

This is a "full table scan" operation but the data sizes are really small so I don't think it's so bad
Then we add that as a schema change while updating dbinit.sql at the same time
New deployments -> Use dbinit.sql, new order
Old deployments -> Use the schema change, land on the same order as dbinit.sql

Reply:
Just a minor note... it doesn't have to be at the end. It'll still work being inserted after `clickhouse` as we intended `clickhouse_keeper` to originally be.

Reply:
I agree with this approach (even if it makes me slightly nervous b/c we have to do a small amount of data migration). There are two primary justifications that lead me to believe this is a good idea

1. Given that enums _are_ ordered, having this integration test is valuable. It's caught this case where migrations result in an effectively different enum shape than what we intended in dbinit
2. We have an active case of database bifurcation. Installations which were or will be provisioned from a schema version greater 4.0.0 will effectively have different values for the `service_kind` and `dataset_kind` than those provisioned with an earlier version that had to run that migration. I don't have a full understanding of what challenges this could cause outside of ordering issues but it seems safer to sand down this sharp edge before we cut ourselves on it.

Reply:
I think I did it in 338828a24837790f7b96a30dde7aecf7963748d0.

See the readme I added for the migration summarizing what it does: https://github.com/oxidecomputer/omicron/blob/338828a24837790f7b96a30dde7aecf7963748d0/schema/crdb/10.0.0/README.md